tests/components/minio/test_minio.py

```diff
@@ -0,0 +1,195 @@
+"""Tests for Minio Hass related code."""
+import asyncio
+import json
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.minio import QueueListener, DOMAIN, CONF_HOST, \
+    CONF_PORT, CONF_ACCESS_KEY, CONF_SECRET_KEY, CONF_SECURE, CONF_LISTEN, \
+    CONF_LISTEN_BUCKET
+from homeassistant.const import EVENT_HOMEASSISTANT_START, \
+    EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.setup import async_setup_component
+
+from tests.components.minio.common import TEST_EVENT
+
+
+async def test_minio_services(hass, caplog):
+    """Test Minio services."""
+    hass.config.whitelist_external_dirs = set('/tmp')
+
+    minio_client_mock = MagicMock()
+
+    with patch('homeassistant.components.minio.Minio') as minio_mock:
+        minio_mock.return_value = minio_client_mock
+
+        await async_setup_component(
+            hass,
+            DOMAIN,
+            {
+                DOMAIN: {
+                    CONF_HOST: 'localhost',
+                    CONF_PORT: '9000',
+                    CONF_ACCESS_KEY: 'abcdef',
+                    CONF_SECRET_KEY: '0123456789',
+                    CONF_SECURE: 'true',
+                }
+            }
+        )
+
+        hass.bus.fire(EVENT_HOMEASSISTANT_START)
+        await hass.async_block_till_done()
+
+        assert 'Setup of domain minio took' in caplog.text
+
+    # Call services
+    await hass.services.async_call(
+        DOMAIN,
+        'put',
+        {
+            'file_path': '/tmp/some_file',
+            'key': 'some_key',
+            'bucket': 'some_bucket'
+        },
+        blocking=True
+    )
+    minio_client_mock.fput_object.assert_called_with(
```

Comment:
Use:
```py
from unittest.mock import call
...
assert minio_client_mock.fput_object.call_args == call(...)
```
Since the mock object isn't specced, a typo in the assert method would be interpretred as an attribute of the mock.